python/mpi_cmake_modules/documentation_builder.py

```diff
@@ -505,8 +505,11 @@ def build_documentation(build_dir, project_source_dir, project_version):
     readme = [
         p.resolve()
         for p in Path(project_source_dir).glob("*")
-        if p.name.lower() == "readme.md"
+        if p.name.lower() in ["readme.md", "readme.rst"]
     ]
+    # sort alphabetically so that "readme.md" is preferred in case both are
+    # found
+    readme = sorted(readme)
     if readme:
         shutil.copy(str(readme[0]), doc_build_dir / "readme.md")
```

Comment:
here that might be an issue? the extension should be preserve.
so something like this might be better?
```
shutil.copy(str(readme[0]), doc_build_dir / str(readme[0].lower())
```

Reply:
Would the following build actually work or does it depend on it being called "readme.md"?  The build system seems to support restructuredText inside a Markdown file, so I intentionally kept the target name as "readme.md" to not cause any issues further done the line.

Reply:
Btw: If I understand the code correctly the behaviour is the same in the cmake version.

Reply:
Oo actually you are totally right!
My bad I forgot about this detail.